src/handle_action.cpp

```diff
@@ -753,24 +755,92 @@ static void smash()
+                        if( glove_coverage < temp_coverage ) {
+                            glove_coverage = temp_coverage;
+                        }
+                    }
+                }
+
+                // If the player has no weapon or has a weapon with 0 bash.
+                // the second case acts like an automatic force unarmed.
+                if( !w->damage_melee( DT_BASH ) ) {
+                    w_bash = false;
+
+                    // Determine hand/foot damage. 90% coverage is the maximum damage reduction.
+                    const int dam_hand = roll_remainder( 5.0 * std::max( 0.0, 1 - glove_coverage / 90.0 ) );
+                    const int dam_foot = roll_remainder( 5.0 * std::max( 0.0, 1 - shoe_coverage / 90.0 ) );
+
+                    // Determine hand/foot with highest hp
+                    bodypart_id  bp_hand_id = ( u.hp_cur[hp_arm_r] > u.hp_cur[hp_arm_l] ) ? bodypart_id( "hand_r" ) :
+                                              bodypart_id( "hand_l" );
+                    bodypart_id bp_foot_id = ( u.hp_cur[hp_leg_r] > u.hp_cur[hp_leg_l] ) ? bodypart_id( "foot_r" ) :
+                                             bodypart_id( "foot_l" );
+
+                    // Pre-determine limb hp after damage is applied.
+                    const int foot_hp_calc = u.hp_cur[u.bp_to_hp( bp_foot_id.obj().token )] - dam_foot;
+                    const int hand_hp_calc = u.hp_cur[u.bp_to_hp( bp_hand_id.obj().token )] - dam_hand;
+
+                    // If we have a 2 handed weapon with 0 bash, use feet.
+                    // Otherwise, use the highest resultant limb hp to smash with.
+                    bool use_hands = w->is_two_handed( u ) ? false : hand_hp_calc > foot_hp_calc;
+
+                    // If the player has a weapon, do not smash with our hands/feet
+                    // if our best option will break our limb. Use weapon instead.
+                    if( !w->is_null() ) {
+                        w_bash = use_hands ? ( !hand_hp_calc > 0 ) : ( !foot_hp_calc > 0 );
+                    }
+
+                    if( !w_bash && dam_hand && dam_foot ) {
+                        u.apply_damage( nullptr, use_hands ? bp_hand_id : bp_foot_id, use_hands ? dam_hand : dam_foot );
+
+                        bool smashed_furn = ( m.has_furn( smashp ) && m.furn( smashp ).obj().bash.str_min != -1 );
+                        add_msg( m_bad, _( "You hurt your %s trying to smash the %s." ),
+                                 use_hands ? _( "hand" ) : _( "foot" ),
+                                 smashed_furn ? m.furnname( smashp ) : m.tername( smashp ) );
+                    }
+                }
+            }
+
+            // Don't damage weapon if we're using hands or feet.
+            u.handle_melee_wear( w_bash ? *w : item() );
+
+            // Calculate stamina off of using hands or feet.
+            const int mod_sta = 2 * u.get_standard_stamina_cost( w_bash ? w : &item() );
```

Comment:
You can't take the address of a temporary object.
```suggestion
            const int mod_sta = 2 * ( w_bash ? u.get_standard_stamina_cost( w ) : 1 ); // or some other value
```

Reply:
Looks like compiler differences between cmake and an updated visual studios.